Potions for Required Items causes Challenges to Error

### Expected behavior

Be able to do `/challenges` without error.

### Observed/Actual behavior

"Internal error occurred."
```
[17:38:00] [Server thread/INFO]: ThunZan issued server command: /island challenges
[17:38:00] [Server thread/ERROR]: null
org.bukkit.command.CommandException: Unhandled exception executing 'island challenges' in world.bentobox.bskyblock.BSkyBlock$1(island)
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:165) ~[PaperCrane-API-1.20.1-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchCommand(CraftServer.java:987) ~[papercrane-1.20.1.jar:git-Paper-"27e1e23"]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleCommand(ServerGamePacketListenerImpl.java:2558) ~[?:?]
	at org.bukkit.craftbukkit.v1_20_R1.entity.CraftPlayer.chat(CraftPlayer.java:705) ~[papercrane-1.20.1.jar:git-Paper-"27e1e23"]
	at it.emmerrei.mycommand.execute.DispatchCommands.ExecuteCommands(DispatchCommands.java:1454) ~[MyCommand.jar:?]
	at it.emmerrei.mycommand.execute.DispatchCommands.PerformMycmdCommands(DispatchCommands.java:1313) ~[MyCommand.jar:?]
	at it.emmerrei.mycommand.execute.DispatchCommands.ExecuteMyCommandCommands(DispatchCommands.java:525) ~[MyCommand.jar:?]
	at it.emmerrei.mycommand.commands.CmdCustomCommands.onCommand(CmdCustomCommands.java:145) ~[MyCommand.jar:?]
	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) ~[PaperCrane-API-1.20.1-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155) ~[PaperCrane-API-1.20.1-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchCommand(CraftServer.java:987) ~[papercrane-1.20.1.jar:git-Paper-"27e1e23"]
	at org.bukkit.craftbukkit.v1_20_R1.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:64) ~[papercrane-1.20.1.jar:git-Paper-"27e1e23"]
	at com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:265) ~[papercrane-1.20.1.jar:?]
	at net.minecraft.commands.Commands.performCommand(Commands.java:324) ~[?:?]
	at net.minecraft.commands.Commands.performCommand(Commands.java:308) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.performChatCommand(ServerGamePacketListenerImpl.java:2359) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChatCommand$21(ServerGamePacketListenerImpl.java:2319) ~[?:?]
	at net.minecraft.util.thread.BlockableEventLoop.lambda$submitAsync$0(BlockableEventLoop.java:59) ~[?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[papercrane-1.20.1.jar:git-Paper-"27e1e23"]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1337) ~[papercrane-1.20.1.jar:git-Paper-"27e1e23"]
	at net.minecraft.server.MinecraftServer.wrapRunnable(MinecraftServer.java:197) ~[papercrane-1.20.1.jar:git-Paper-"27e1e23"]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1314) ~[papercrane-1.20.1.jar:git-Paper-"27e1e23"]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1307) ~[papercrane-1.20.1.jar:git-Paper-"27e1e23"]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
	at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1285) ~[papercrane-1.20.1.jar:git-Paper-"27e1e23"]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1173) ~[papercrane-1.20.1.jar:git-Paper-"27e1e23"]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:317) ~[papercrane-1.20.1.jar:git-Paper-"27e1e23"]
	at java.lang.Thread.run(Thread.java:840) ~[?:?]
Caused by: java.lang.NoSuchMethodError: 'java.util.List org.bukkit.potion.PotionType.getPotionEffects()'
	at world.bentobox.bentobox.hooks.LangUtilsHook.getPotionBaseEffectName(LangUtilsHook.java:370) ~[BentoBox-2.4.1.jar:?]
	at world.bentobox.challenges.utils.Utils.prettifyObject(Utils.java:683) ~[Challenges-1.3.0.jar:?]
	at world.bentobox.challenges.utils.Utils.prettifyObject(Utils.java:718) ~[Challenges-1.3.0.jar:?]
	at world.bentobox.challenges.utils.Utils.prettifyObject(Utils.java:583) ~[Challenges-1.3.0.jar:?]
	at world.bentobox.challenges.panel.CommonPanel.lambda$generateInventoryChallenge$7(CommonPanel.java:371) ~[Challenges-1.3.0.jar:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) ~[?:?]
	at java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:510) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?]
	at world.bentobox.challenges.panel.CommonPanel.generateInventoryChallenge(CommonPanel.java:362) ~[Challenges-1.3.0.jar:?]
	at world.bentobox.challenges.panel.CommonPanel.generateRequirements(CommonPanel.java:268) ~[Challenges-1.3.0.jar:?]
	at world.bentobox.challenges.panel.CommonPanel.generateChallengeDescription(CommonPanel.java:148) ~[Challenges-1.3.0.jar:?]
	at world.bentobox.challenges.panel.user.ChallengesPanel.createChallengeButton(ChallengesPanel.java:275) ~[Challenges-1.3.0.jar:?]
	at world.bentobox.challenges.panel.user.ChallengesPanel.createChallengeButton(ChallengesPanel.java:245) ~[Challenges-1.3.0.jar:?]
	at world.bentobox.bentobox.api.panels.TemplatedPanel.makeAddonButton(TemplatedPanel.java:295) ~[BentoBox-2.4.1.jar:?]
	at world.bentobox.bentobox.api.panels.TemplatedPanel.makeButton(TemplatedPanel.java:248) ~[BentoBox-2.4.1.jar:?]
	at world.bentobox.bentobox.api.panels.TemplatedPanel.processItemData(TemplatedPanel.java:126) ~[BentoBox-2.4.1.jar:?]
	at world.bentobox.bentobox.api.panels.TemplatedPanel.populateInventoryPanel(TemplatedPanel.java:81) ~[BentoBox-2.4.1.jar:?]
	at world.bentobox.bentobox.api.panels.TemplatedPanel.generatePanel(TemplatedPanel.java:61) ~[BentoBox-2.4.1.jar:?]
	at world.bentobox.bentobox.api.panels.TemplatedPanel.<init>(TemplatedPanel.java:52) ~[BentoBox-2.4.1.jar:?]
	at world.bentobox.bentobox.api.panels.builders.TemplatedPanelBuilder.build(TemplatedPanelBuilder.java:150) ~[BentoBox-2.4.1.jar:?]
	at world.bentobox.challenges.panel.user.ChallengesPanel.build(ChallengesPanel.java:107) ~[Challenges-1.3.0.jar:?]
	at world.bentobox.challenges.panel.user.ChallengesPanel.open(ChallengesPanel.java:70) ~[Challenges-1.3.0.jar:?]
	at world.bentobox.challenges.commands.ChallengesPlayerCommand.execute(ChallengesPlayerCommand.java:87) ~[Challenges-1.3.0.jar:?]
	at world.bentobox.bentobox.api.commands.CompositeCommand.call(CompositeCommand.java:292) ~[BentoBox-2.4.1.jar:?]
	at world.bentobox.bentobox.api.commands.CompositeCommand.execute(CompositeCommand.java:260) ~[BentoBox-2.4.1.jar:?]
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155) ~[PaperCrane-API-1.20.1-R0.1-SNAPSHOT.jar:?]
	... 31 more
```

### Steps/models to reproduce

1. Make or edit existing challenge
2. Go to required items
3. Add potions to required items (standard potions like resistance 1 potion etc)
4. Save
5. Do `/challenges`
6. Error

### BentoBox version

BB Version 2.4.1
Challenges 1.3.0

### Plugin list

N/A

### Other

For future reference, it should have a fail case where it replaced it with a replacement item. It shouldn't error and cause the whole command to not work.

--- a/src/main/java/world/bentobox/bentobox/hooks/LangUtilsHook.java
+++ b/src/main/java/world/bentobox/bentobox/hooks/LangUtilsHook.java
@@ -363,16 +363,23 @@ public class LangUtilsHook extends Hook {
      * @param user       The user's language will be used for translation.
      * @return Return the translation result.
      */
+    @SuppressWarnings("deprecation")
     public static String getPotionBaseEffectName(PotionType potionType, User user) {
         if (hooked) {
             return LanguageHelper.getPotionBaseEffectName(potionType, getUserLocale(user));
         }
-        List<PotionEffect> effects = potionType.getPotionEffects();
-        if (effects.isEmpty()) {
-            return "No Effects";
+        try {
+            List<PotionEffect> effects = potionType.getPotionEffects();
+
+            if (effects.isEmpty()) {
+                return "No Effects";
+            }
+            return effects.stream().map(effect -> Util.prettifyText(effect.getType().getKey().getKey()))
+                    .collect(Collectors.joining(", "));
+        } catch (Exception e) {
+            // Older versions of Spigot pre-1.20.4 don't have getPotionEffects()
+            return Util.prettifyText(potionType.getEffectType().getKey().getKey());
         }
-        return effects.stream().map(effect -> Util.prettifyText(effect.getType().getKey().getKey()))
-                .collect(Collectors.joining(", "));
     }
 
     /**

